src/accumulator.jl

```diff
@@ -61,32 +56,80 @@ done(ct::Accumulator, state) = done(ct.map, state)
 
 # manipulation
 
-push!(ct::Accumulator, x, a::Number) = (ct.map[x] = ct[x] + a)
-push!(ct::Accumulator{T,V}, x) where {T,V} = push!(ct, x, one(V))
+"""
+    inc!(ct, x, [v=1])
+
+Increments the count for `x` by `v` (defaulting to one)
+"""
+inc!(ct::Accumulator, x, a::Number) = (ct[x] += a)
+inc!(ct::Accumulator{T,V}, x) where {T,V} = inc!(ct, x, one(V))
+
+# inc! is preferred over push!, but we need to provide push! for the Bag interpreation
+# which is used by classified_collections.jl
+push!(ct::Accumulator, x) = inc!(ct, x)
+push!(ct::Accumulator, x, a::Number) = inc!(ct, x, a)
 
 # To remove ambiguities related to Accumulator now being a subtype of Associative
-push!(ct::Accumulator{T,V}, x::T) where T<:Pair where V = push!(ct, x, one(V))
-push!(ct::Accumulator{T,V}, x::Pair) where {T,V} = push!(ct, convert(T, x))
+push!(ct::Accumulator, x::Pair)  = inc!(ct, x)
+
+
+
+"""
+    dec!(ct, x, [v=1])
 
-function push!(ct::Accumulator, r::Accumulator)
-    for (x, v) in r
-        push!(ct, x, v)
+Decrements the count for `x` by `v` (defaulting to one)
+"""
+dec!(ct::Accumulator, x, a::Number) = (ct[x] -= a)
+dec!(ct::Accumulator{T,V}, x) where {T,V} = dec!(ct, x, one(V))
```

Comment:
Okay to leave as is, but I think this can just be 

```julia
dec!(ct::Accumulator, x, a::Number = 1) = (ct[x] -= a)
```

I guess it might be worth looking at the code generated for each of these in different cases, but in general, until it is shown to be better to create more complicated code, I think simpler code is better.